drift/lib/src/runtime/manager/references.dart

```diff
@@ -0,0 +1,491 @@
+part of 'manager.dart';
+
+/// A simple function for generating aliases for referenced columns
+///
+/// This function is used internally by generated code and should not be used directly.
+// ignore: non_constant_identifier_names
+String $_aliasNameGenerator(
+    GeneratedColumn currentColumn, GeneratedColumn referencedColumn) {
+  return '${currentColumn.tableName}__${currentColumn.name}__${referencedColumn.tableName}__${referencedColumn.name}';
+}
+
+/// Base class for the "WithReference" classes
+///
+/// When a user calls `withReferences` on a manager, the item is returned, along with an instance of this
+/// class which contains getters for the referenced tables manager which are pre-filtered to the item.
+/// So the following:
+/// ```dart
+/// final (group,refs) = await groups.filter((f) => f.id(5)).withReferences().getSingle()
+/// final usersInGroup = refs.users.get() // filter((f) => f.group(group.id)) is already applied
+/// ```
+/// is short for:
+/// ```dart
+/// final group = await groups.filter((f) => f.id(5)).getSingle()
+/// final usersInGroup = await users.filter((f) => f.group(group.id)).get()
+/// ```
+///
+/// {@macro manager_internal_use_only}
+///
+/// # Prefetching
+///
+/// ### Problem
+///
+/// It's quite common that a user will want a model together with it's relations.
+/// The simple option is to prefetch all of the users, and then run a 2nd request for each related group.
+/// ```dart
+/// final users = await users.withReferences().get()
+/// for (final (user,refs) in users){
+///   final group = await refs.groups.getSingle();
+/// }
+/// ```
+/// The issue with the above code is that we could be setting ourselves up to do tons of queries.
+///
+/// ### Solution
+///
+/// There are two methods we use together to solve this.
+/// Joins and Prefetches.
+///
+/// ##### Join
+///
+/// For fields which are foreign keys to other tables, we can just use a join to select the related field and get it all in one query
+/// There are many examples how to do this in the drift documentation
+///
+/// ##### Prefetch
+///
+/// However, for reverse relations, we can't get it all in a single query.
+/// We have to run 2 queries. Here is how it's done manually.
+///
+/// ```dart
+/// final groups = await groups.get();
+/// final groupIds = groups.map((group)=> group.id);
+/// final users = await users.filter((f)=> f.group.id.isIn(groupIds)).get()
+/// final groupsWithUsers = groups.map((group)=>(group,users.where((user)=> user.group == group.id)));
+/// ```
+///
+/// We get all the users ahead of time, and then manually return each group with it's users using a `where(...)` filter.
+///
+/// We will call both of these "prefetch" throughout.
+///
+///
+/// ### Manager API
+///
+/// This is quite verbose, and the manager api seeks to solve this for you.
```

Comment:
This section is aimed at contributors, so

```suggestion
/// This is quite verbose, and the manager api seeks to solve this for drift users.
```